artwork/Modules/Project/Services/ProjectService.php

```diff
@@ -42,6 +42,32 @@ public function getProjectByCostCenter(string $costCenter): Project|null
         return $this->projectRepository->getProjectByCostCenter($costCenter);
     }
 
+    public function getProjects(): \Illuminate\Database\Eloquent\Builder
+    {
+        return Project::with([
```

Comment:
Das bitte noch in ein eigenes Repository, dann haben wir dafür eine grundlegende Funktion die wir in jedem Service nutzen können.

Darüber hinaus ist dann für das Repository noch ein Unit-Test erforderlich, der assertet wie das Query zusammengebaut wird. Also wie oft wird whereNull, orderBy, without, (und mit welchen Parametern) aufgerufen auf dem Builder. Dazu musst du das Model mocken und assertest genau wie das array aussieht was du an die with Methode übergibst. Das with returned dann wiederrum ein gemockten Builder auf dem du die funktionsaufrufe asserten musst.

Assert Project ruft "with" auf mit Array X (assert die keys, assert die values)
Assert Builder ruft whereNull mit pinney_by_users auf
Assert Builder ruft orderBy mit id und DESC auf
Assert Builder ruft without mit array(shiftRelevantEventTypes) auf